Shell/Shell.cpp

```diff
@@ -442,6 +442,65 @@ int Shell::builtin_fg(int argc, const char** argv)
     return return_value;
 }
 
+int Shell::builtin_disown(int argc, const char** argv)
+{
+    Vector<const char*> str_job_ids;
+
+    Core::ArgsParser parser;
+    parser.add_positional_argument(str_job_ids, "Id of the jobs to disown (omit for current job)", "job_ids", Core::ArgsParser::Required::No);
+
+    if (!parser.parse(argc, const_cast<char**>(argv), false))
+        return 1;
+
+    Vector<size_t> job_ids;
+    for (auto& job_id : str_job_ids) {
+        bool ok;
+        auto id = StringView { job_id }.to_uint(ok);
+        if (ok)
+            job_ids.append(id);
+        else
+            printf("Invalid job id: %s\n", job_id);
+    }
+
+    if (job_ids.is_empty())
+        job_ids.append(jobs.size() - 1);
+
+    Vector<size_t> keys_of_jobs_to_disown;
+
+    for (auto id : job_ids) {
+        bool found = false;
+        for (auto& entry : jobs) {
+            if (entry.value->job_id() == id) {
+                keys_of_jobs_to_disown.append(entry.key);
+                found = true;
+                break;
+            }
+        }
+        if (!found) {
+            printf("job with id %zu not found\n", id);
+        }
+    }
+    if (keys_of_jobs_to_disown.is_empty()) {
+        if (str_job_ids.is_empty()) {
+            printf("disown: no current job\n");
+        }
+        // An error message has already been printed about the nonexistence of each listed job.
+        return 1;
+    }
+    for (auto job_index : keys_of_jobs_to_disown) {
+        auto job = jobs.get(job_index).value();
+
+        job->deactivate();
+
+        if (!job->is_running_in_background())
+            printf("disown warning: job %llu is currently not running, 'kill -%d %d' to make it continue\n", job->job_id(), SIGCONT, job->pid());
```

Comment:
I _really_ want this "`kill -%d %d`" to be a link, so let's talk specifics about how this would work (summoning @bugaevc)

- the command _must_ be run in the invoking shell
- it would be nice to get its output
- writing the command into the stdin of the shell is not particularly wise

I think that json-escape-sequence-xpc is gonna have to come sooner than later...

Reply:
XPC?? nonononono

Reply:
> writing the command into the stdin of the shell is not particularly wise

maybe it is...maybe clicking a command link should just write it to stdin, and the user should be expected to know this?

Reply:
> nonononono

lol

Reply:
You meant IPC, RPC, or what? :slightly_smiling_face:

Reply:
I legit meant cross-process communication there.
it waz bait to get you here :)

Reply:
apparently that's an apple-specific term?
I dunno, I've only had use of it with objc, so I called it xpc

Reply:
aaaaaaaaaaaanyway, should we just make clicking a link to a command write it to the stdin of the shell (plus a newline)?

Reply:
![shot-2020-05-25_00-38-10](https://user-images.githubusercontent.com/14001776/82763878-0a780000-9e20-11ea-9cfa-fce54f20c8a9.png)

:thinking: